pkg/user/user.go

```diff
@@ -118,3 +120,12 @@ func (u *User) RemoveAuthByProvider(provider string) bool {
 func (u *User) ClearAuths() {
 	u.auths = []Auth{}
 }
+
+func EncodePassword(pass string) ([]byte, error) {
```

Comment:
The password encoding process should be completely confined within the domain model and completely invisible from the outside. The use case layer should not know how the password is encoded at all. This is a security issue and should be implemented carefully.

`EncodePassoword` func should be private and used only by `func (*User) SetPassword(password string)` setter method. Also `comparePassword` function is needed and should be called by `func (*User) ComparePassword(password string) bool` func. 

```suggestion
func comparePassword(pass, pass2 string) bool {
	// ...
}

func cncodePassword(pass string) ([]byte, error) {
```